src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -241,10 +245,50 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
                     }
                 }
             }
-
             return _longRunningOperations;
         }
 
+        private Dictionary<Operation, NonLongRunningOperation> EnsureNonLongRunningOperations()
+        {
+            if (_nonLongRunningOperations != null)
+            {
+                return _nonLongRunningOperations;
+            }
+
+            _nonLongRunningOperations = new Dictionary<Operation, NonLongRunningOperation>();
+            var desiredHttpMethods = new HttpMethod[] { HttpMethod.Put, HttpMethod.Delete, HttpMethod.Patch };
+
+            if (_context.Configuration.PublicClients)
+            {
+                foreach (var operationGroup in _codeModel.OperationGroups)
+                {
+                    // if non-resource, we won't be able to get the info for LRO
+                    if (operationGroup.IsResource(_mgmtConfiguration))
+                    {
+                        foreach (var operation in operationGroup.Operations)
```

Comment:
operationGroup has an extension method to get the list of operations by httpmethod as an index which would make this more efficient, it takes care of the operation.Requests.FirstOrDefault check as well as the desiredHttpMethods.contains check

var operationList = operationGroup.OperationHttpMethodMapping[HttpMethod.Put]